src/python-imports-search.ts

```diff
@@ -1,67 +1,91 @@
-import { EMPTY, from, of } from 'rxjs';
-import { concatMap, toArray } from 'rxjs/operators';
+import { from, Subscription } from 'rxjs'
+import { filter, map, switchMap } from 'rxjs/operators'
 import * as sourcegraph from 'sourcegraph'
+import { resolveSettings, Settings } from './settings'
+
+const decorationType = sourcegraph.app.createDecorationType && sourcegraph.app.createDecorationType()
+const settings = resolveSettings(sourcegraph.configuration.get<Settings>().value)
 
 export function activate(): void {
-   sourcegraph.search.registerQueryTransformer({
-       transformQuery: (query: string) => {
-           const pyImportsRegex = /\bpy.imports:([^\s]*)/i
-           if (query.match(pyImportsRegex)) {
-               const pyImportsFilter = query.match(pyImportsRegex)
-               const pyPkg = pyImportsFilter && pyImportsFilter.length >= 1 ? pyImportsFilter[1] : ''
-               const singleImport = '^import\\s' + pyPkg + '$'
-               const namedImport = '^import\\s' + pyPkg + '\\sas\\s[^\\s]*'
-               const scopedImport = '^from\\s(.*)\\simport\\s(.*)' + pyPkg
-               return query.replace(pyImportsRegex  , `(${singleImport}|${namedImport}|${scopedImport}) lang:python `)
-           }
-           return query
-        }
-   })
+    const subscriptions = new Subscription()
```

Comment:
Please use [`ExtensionContext`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@5e9f971/-/blob/packages/sourcegraph-extension-api/src/sourcegraph.d.ts#L1368-1399) (passed to the `activate` function) for this.

Reply:
https://github.com/sourcegraph/python-imports-search/commit/232b7aaa84eff52dc9f4ea4396b49724da0656f4

Reply:
Huh. I did that thing where I clicked the GitHub email review request link and it took me to a subset of the changes. Sorry!